src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalStatistics.cs

```diff
@@ -0,0 +1,249 @@
+
+        // Not sure how to determine this.
+        public override int NumberOfIPAddresses
+        {
+            get
+            {
+                int count = 0;
+                foreach (LinuxNetworkInterface lni in LinuxNetworkInterface.GetLinuxNetworkInterfaces())
+                {
+                    // Could there be duplicates, somehow?
+                    count += lni.Addresses.Count;
+                }
+                return count;
+            }
+        }
+
+        public override int NumberOfRoutes { get { return _numRoutes; } }
+
+        public override long OutputPacketRequests { get { return _outRequests; } }
+
+        /* Should be:
+
+            ipRoutingDiscards OBJECT-TYPE
+              SYNTAX Counter
+              ACCESS read-only
+              STATUS mandatory
+              DESCRIPTION
+                  "The number of routing entries which were chosen
+                  to be discarded even though they are valid.  One
+                  possible reason for discarding such an entry could
+                  be to free - up buffer space for other routing
+
+            Isn't exposed in snmp or snmp6, although it is listed as mandatory in the specification.
+        */
+        public override long OutputPacketRoutingDiscards { get { throw new PlatformNotSupportedException(); } }
+
+        public override long OutputPacketsDiscarded { get { return _outDiscards; } }
+
+        public override long OutputPacketsWithNoRoute { get { return _outNoRoutes; } }
+
+        public override long PacketFragmentFailures { get { return _fragmentFails; } }
+
+        public override long PacketReassembliesRequired { get { return _reassemblyRequireds; } }
+
+        public override long PacketReassemblyFailures { get { return _reassemblyFails; } }
+
+        public override long PacketReassemblyTimeout { get { return _reassemblyTimeout; } }
+
+        public override long PacketsFragmented { get { return _fragmentCreates; } }
+
+        public override long PacketsReassembled { get { return _reassemblyOKs; } }
+
+        public override long ReceivedPackets { get { return _inReceives; } }
+
+        public override long ReceivedPacketsDelivered { get { return _inDelivers; } }
+
+        public override long ReceivedPacketsDiscarded { get { return _inDiscards; } }
+
+        public override long ReceivedPacketsForwarded { get { return _forwardedDatagrams; } }
+
+        public override long ReceivedPacketsWithAddressErrors { get { return _inAddressErrors; } }
+
+        public override long ReceivedPacketsWithHeadersErrors { get { return _inHeaderErrors; } }
+
+        public override long ReceivedPacketsWithUnknownProtocol { get { return _inUnknownProtocols; } }
+
+        private static string ReadProcConfigFile(string v)
+        {
+            return File.ReadAllText(v);
+        }
+
+        private static int CountOccurences(string value, string candidate)
+        {
+            int index = 0;
```

Comment:
We should probably check that the `candidate` is not null. 
If you believe it will always be guaranteed by the caller then we can have a `Debug.Assert`.